drivers/pds/pdsbackup/backuputils.go

```diff
@@ -0,0 +1,235 @@
+		return nil, fmt.Errorf("Error in adding the backup credentials to PDS , Err: %v ", err)
+	}
+	log.Info("Create google storage.")
+	err = backupClient.gcpStorageClient.createBucket(bucketName)
+	if err != nil {
+		return nil, fmt.Errorf("Failed while creating bucket, Err: %v ", err)
+	}
+	log.Infof("Adding backup target {Name: %v} to PDS.", name)
+	backupTarget, err := backupClient.components.BackupTarget.CreateBackupTarget(tenantId, name, backupCred.GetId(), bucketName, "", "google")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create google backup target, Err: %v ", err)
+	}
+	return backupTarget, nil
+
+}
+
+// DeleteAwsS3BackupCredsAndTarget delete backup creds,bucket and target.
+func (backupClient *BackupClient) DeleteAwsS3BackupCredsAndTarget(backupTargetId string) error {
+	log.Info("Removing S3 backup creadentials and target from PDS.")
+	backupTarget, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+	credId := backupTarget.GetBackupCredentialsId()
+	log.Infof("Deleting backup target {Name: %v} to PDS.", backupTarget.GetName())
+	_, err := backupClient.components.BackupTarget.DeleteBackupTarget(backupTargetId)
+	if err != nil {
+		return fmt.Errorf("Failed to delete AWS S3 backup target, Err: %v ", err)
+	}
+	waitErr := wait.Poll(bkpTimeInterval, bkpMaxtimeInterval, func() (bool, error) {
+		model, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+		if model != nil {
+			log.Info(model.GetName())
+			return false, nil
+		} else {
+			return true, nil
+		}
+	})
+	if waitErr != nil {
+		return fmt.Errorf("error occured while polling for deleting backuptarget : %v", err)
+	}
+	_, err = backupClient.components.BackupCredential.DeleteBackupCredential(credId)
+	if err != nil {
+		return fmt.Errorf("Failed to delete AWS S3 backup credentials, Err: %v ", err)
+	}
+	log.Info("Delete S3 bucket from AWS cloud.")
+	err = backupClient.awsStorageClient.deleteBucket(bucketName)
+	if err != nil {
+		return fmt.Errorf("Failed to delete S3 bucket %s, Err: %v ", bucketName, err)
+	}
+	return nil
+}
+
+// DeleteAzureBackupCredsAndTarget delete backup creds,bucket and target.
+func (backupClient *BackupClient) DeleteAzureBackupCredsAndTarget(backupTargetId string) error {
+	log.Info("Removing Azure backup creadentials and target from PDS.")
+	backupTarget, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+	credId := backupTarget.GetBackupCredentialsId()
+	log.Infof("Deleting backup target {Name: %v} to PDS.", backupTarget.GetName())
+	_, err := backupClient.components.BackupTarget.DeleteBackupTarget(backupTargetId)
+	backupTarget.GetBackupCredentialsId()
+	if err != nil {
+		return fmt.Errorf("Failed to delete Azure backup target, Err: %v ", err)
+	}
+	waitErr := wait.Poll(bkpTimeInterval, bkpMaxtimeInterval, func() (bool, error) {
+		model, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+		if model != nil {
+			log.Info(model.GetName())
```

Comment:
can you please check this logic, it is returning nil for all cases

Reply:
Done